File: play-core/src/main/kotlin/play/sys.java
package play;

import java.io.File;

public final class sys {
    private sys() {
    }

    public static Runtime runtime() {
        return Runtime.getRuntime();
    }

    public static int availableProcessors() {
        return runtime().availableProcessors();
    }

    public static boolean isWindows() {
        return isOS("Windows");
    }

    public static boolean isMac() {
        return isOS("Mac OS");
    }

    public static boolean isLinux() {
        return isOS("Linux");
    }

    private static boolean isOS(String osName) {
        return SystemProperties.osName().startsWith(osName);
    }

    public String fileSeparator() {
        return File.separator;
    }

    public String lineSeparator() {
        return System.lineSeparator();
    }
}

File: play-core/src/main/kotlin/play/sys.kt
package play

import java.io.File
import play.SystemProps.osName

object sys {

  init {
    SystemProps.setIfAbsent("availableProcessors", runtime().availableProcessors())
  }

  @JvmStatic
  fun runtime(): Runtime {
    return Runtime.getRuntime()
  }

  @JvmStatic
  fun availableProcessors(): Int {
    return SystemProps.getInt("availableProcessors")
  }

  @JvmStatic
  val isWindows: Boolean
    get() = isOS("Windows")

  @JvmStatic
  val isMac: Boolean
    get() = isOS("Mac OS")

  @JvmStatic
  val isLinux: Boolean
    get() = isOS("Linux")

  private fun isOS(osName: String): Boolean {
    return osName().startsWith(osName)
  }

  @JvmStatic
  fun fileSeparator(): String {
    return File.separator
  }

  @JvmStatic
  fun pathSeparator(): String {
    return File.pathSeparator
  }

  @JvmStatic
  fun lineSeparator(): String {
    return System.lineSeparator()
  }
}
